Formula/scotch.rb

```diff
@@ -46,8 +55,12 @@ def install
         return 0;
       }
     EOS
-
     system ENV.cc, "test.c", "-L#{lib}", "-lscotch"
     assert_match version.to_s, shell_output("./a.out")
+
+    # PT-scotch library test
+    system "mpicc", pkgshare/"test_strat_par.c", "-o", "test_strat_par",
+           "-I#{include}", "-L#{lib}", "-lptscotch", "-lscotch", "-lptscotcherr", "-lm", "-pthread"
+    assert_match "Parallel mapping strategy, SCOTCH_STRATDEFAULT", shell_output("./test_strat_par")
```

Comment:
Why remove `test_strat.c` for `scotch`?

Reply:
I thought that you want to keep the previous test for scotch, so I removed mine.

Reply:
I did want to keep it, but there's no need to remove the test you added.

Reply:
I hope this is the last commit so I can squash it. The missing pt-scotch issue should be fixed a long time ago.
The PR #76678 that removes `pt-scotch` got passed so easily. And the effort to bring it back is being reviewed so strictly.

Reply:
I'm not sure why you think #76678 removes `pt-scotch`. `scotch` didn't exist in Homebrew/core at the time that PR was made, so there was no `pt-scotch` to remove.

In any case, this looks good now. Thanks for all your work here! No need to squash this -- @BrewTestBot knows how to squash version bump PRs (but is less smart about more complicated changes).

Reply:
> I'm not sure why you think #76678 removes `pt-scotch`. `scotch` didn't exist in Homebrew/core at the time that PR was made, so there was no `pt-scotch` to remove.

The history shows that `scotch` existed in Homebrew/core in 2013 (before it got moved to Homebrew/science)
https://github.com/Homebrew/homebrew-core/blob/6095bab4530021e426f2e547618acaf124e53604/Formula/scotch.rb#L23

Reply:
It did exist previously, and was removed from Homebrew/core when it was moved to Homebrew/science. What the formula was a long time ago doesn't have any bearing on what it should look like when added again as a new formula.